cmd/contour/contour.go

```diff
@@ -34,10 +34,11 @@ func main() {
 	envoyCmd := app.Command("envoy", "Sub-command for envoy actions.")
 	sdm, shutdownManagerCtx := registerShutdownManager(envoyCmd, log)
 
+	bootstrap, bootstrapCtx := registerBootstrap(app)
+
 	// Add a "shutdown" command which initiates an Envoy shutdown sequence.
-	sdmShutdown, sdmShutdownCtx := registerShutdown(envoyCmd, log)
+	sdmShutdown, sdmShutdownCtx := registerShutdown(envoyCmd, bootstrapCtx.AdminPort, log)
```

Comment:
The `bootstrapCtx.AdminPort` arg should come out and need to use the flag defined in the shutdown manager command: https://github.com/projectcontour/contour/pull/3501/files#diff-ab19309345ae701385f9f0a7315157a1a76ad909f3dacef2e8ad9fb973a4232fR296

Reply:
Hi @stevesloka I'm not getting what you mean. That param `bootstrapCtx.AdminPort` is exactly the flag (admin-port) specified in the shutdown manager command. So what is the change?

Reply:
The fact is, that if I remove that param, practically I revert all my changes, leaving the code the way it was before and leaving the original issue unresolved.

Reply:
Hi, @stevesloka thanks for the kind suggestions. I've applied all the changes you told me. Hope it's good